### PR TITLE
add duplicate check to toctree_fix function

### PR DESCRIPTION
**Pull request by mawieland**
fix for wrongly assigned page number in table of content in case of duplicated headlines

### CODE REVIEW

File: sphinx_simplepdf/builders/simplepdf.py [hunk 1]
```diff
@@ -172,6 +173,45 @@ def _toctree_fix(self, html):
             for link in links:
                 link["href"] = link["href"].replace(f"{self.app.config.root_doc}.html", "")
 
+            print(links)
```
Comment (danwos) [inline sphinx_simplepdf/builders/simplepdf.py:176]: are the print() statements needed?

File: sphinx_simplepdf/builders/simplepdf.py [hunk 2]
```diff
@@ -172,6 +173,45 @@ def _toctree_fix(self, html):
...
+
+            for text, counter in duplicates.items():
+
+                ref = re.findall("href=\"#.*\"", str(text))
+                
+                # clean href data for searching
+                cleaned_ref_toc = ref[0].replace("href=\"", "").replace("\"", "") # "#target"
+                cleaned_ref_target = ref[0].replace("href=\"#", "").replace("\"", "") # "target"
+
+                # occurences = soup.find_all('a', attrs={"class": "headerlink", "href": cleaned_ref})
```
Comment (danwos) [inline sphinx_simplepdf/builders/simplepdf.py:194]: please remove not needed comments

**Review by danwos — CHANGES_REQUESTED** [review]
Thanks for the PR.
Overall looks good :+1: 
Just 1-2 small points for cleaner code :)

File: sphinx_simplepdf/builders/simplepdf.py (hunk 1 above)
Comment (mawieland) [reply sphinx_simplepdf/builders/simplepdf.py]: I left one to tell the user if a duplicate headline has been found. others have been removed

File: sphinx_simplepdf/builders/simplepdf.py (hunk 2 above)
Comment (mawieland) [reply sphinx_simplepdf/builders/simplepdf.py]: done

**Review by danwos — APPROVED** [review]
Looks good to me!
Thanks for the PR :+1:

**Pull request merged**